### DISCUSSION TITLE
Failed to start java agent

### DISCUSSION THREAD

**Discussion by kk088635** [Ginger-Automation/Ginger#3393] [Q&A]
I am currently working on a PoC on a Java application. When I try to start the java agent, I get an error message " Failed to load agent, Error Details: 'Please confirm Agent configurations are valid'.
My application is running in the same machine as Amdocs ginger.

Error Logs : 
Amdocs.Ginger.CoreNET.log4netLib.GingerLog |:
Method - <ConnectToJavaDriver>b__45_0, Error - No connection could be made because the target machine actively refused it. [::ffff:127.0.0.1]:8888
System.Net.Internals.SocketExceptionFactory+ExtendedSocketException (10061): No connection could be made because the target machine actively refused it. [::ffff:127.0.0.1]:8888
   at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)
   at System.Net.Sockets.TcpClient.Connect(IPEndPoint remoteEP)
   at GingerCore.Drivers.JavaDriverLib.JavaDriver.<ConnectToJavaDriver>b__45_0() in D:\BuildConfigs\tfsagent\_work\4\s\Ginger\GingerCore\Drivers\JavaDriverLib\JavaDriver.cs:line 239

Platform: Windows
Agent Configuration: JavaAgentHost : 127.0.0.1, JavaAgentPort : 8888

Please could you someone help me with this issue.

**Comment by jonasermert** [comment]
Make sure that there are no other applications or processes using the same port (8888). You can use tools like netstat to check for open ports and identify any conflicts. Open a command prompt and run: netstat -an | find "8888"

**Reply by udaync-evoke** [reply]
@jonasermert @JinendraGhodke , Port 8888 is not being used by any of the processes, still getting error message as 
Method - <ConnectToJavaDriver>b__45_0, Error - No connection could be made because the target machine actively refused it. [::ffff:127.0.0.1]:8888
System.Net.Sockets.SocketException (10061): No connection could be made because the target machine actively refused it. [::ffff:127.0.0.1]:8888

**Comment by GokulBothe** [comment]
@udaync-evoke ,

Automate Java application
please follow the below steps-

**Create Ginger Solution:** Include the Java target application and the Java agent.
**Add Launch Action:** From the Actions Library, go to the General Actions tab and add the "Launch Java Application" action.
**Open Operation Settings:** Access the Operation Settings for the "Launch Java Application" action.
**Select Java Version Path:** Choose the Java version path that matches your application.
**Provide Application Path:** Enter the path to your Java application.
**Configure Waiting Options:**

1. Check "Wait for Java Application Window".
2. Provide the full or partial title of the Java window.
3. Set the maximum waiting time or keep it default.

**Attach Java Agent:**

1. Check "Attach Ginger Java Agent".
2. Keep the default settings.

**Run the Action:** Click the "Run Action" button and wait for the Java application to launch.
**Start Agent:**

1. At the top of the Automate screen, click the Agent Start toggle button.
2. Ensure it turns green, indicating it has started.

**Create POM:** You can now create a Page Object Model (POM) of the Java application and begin automation.
**Automation Options:**

1. Use the Record option.
2. Utilize Live Spy.
3. Explore additional automation options.

for more detail refer below image

![image](https://github.com/Ginger-Automation/Ginger/assets/123472538/1da60e5b-936b-457d-b90b-ff549c90bab0)

@JinendraGhodke
